src/openlmis-quantity-unit-toggle/quantity-unit.service.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+(function() {
+
+    'use strict';
+
+    /**
+   * @ngdoc service
+   * @name openlmis-quantity-unit-toggle.quantityUnitCalculateService
+   *
+   * @description
+   * Responsible for calculations on olmis related quantities
+   */
+    angular
+        .module('openlmis-quantity-unit-toggle')
+        .service('quantityUnitCalculateService', service);
+
+    service.$inject = [];
+
+    function service() {
+
+        this.recalculateSOHQuantity = recalculateSOHQuantity;
+        this.recalculateInputQuantity = recalculateInputQuantity;
+
+        /**
+         * @ngdoc method
+         * @methodOf openlmis-quantity-unit-toggle.quantityUnitCalculateService
+         * @name recalculateSOHQuantity
+         *
+         * @description
+         * Recalculates the given stock on hand quantity to packs or to doses
+         *
+         * @param  {Object}  stockOnHand   the quantity to be recalculated
+         * @param  {number}  netContent    quantity of doses in one pack
+         * @param  {boolean} inDoses       unit to recalculated
+         * 
+         * @return {String}                quantity in the appropriate unit (packs or doses)
+         */
+        function recalculateSOHQuantity(stockOnHand, netContent, inDoses) {
+            if (inDoses) {
+                return stockOnHand;
+            }
+            var packs = (stockOnHand > 0) ? Math.floor(stockOnHand / netContent) : Math.ceil(stockOnHand / netContent);
+            var remainderDoses = stockOnHand % netContent;
+
+            if (remainderDoses === 0) {
+                return packs;
+            }
+
+            if (remainderDoses > 0) {
+                return packs + ' ( +' + remainderDoses + ' doses)';
+            }
+            return packs + ' ( ' + remainderDoses + ' doses)';
+        }
```

Comment:
1. `netContent` can be 0/negative? If so, we should handle it properly? We can check if its positive and if its a number.
2. What about hardcoded `doses` in here? We won't be able to translate it.

Reply:
I checked and unfortunately the system allows netContent to be 0. I added corrections